src/libcoretest/char.rs

```diff
@@ -296,3 +296,21 @@ fn eu_iterator_specializations() {
     check('\u{12340}');
     check('\u{10FFFF}');
 }
+
+#[test]
+fn decode_utf8() {
+    fn eq_iter<T: PartialEq, I: Iterator<Item = T>, J: Iterator<Item = T>>(i: I, j: J) -> bool {
+        Iterator::zip(i, j).all(|(x, y)| x == y)
+    }
```

Comment:
The `zip` adapter ends up being the shorter of the two lengths, so this may want to ensure that the iterators have the same length as well